lib/docker-toolbelt.coffee

```diff
@@ -102,6 +102,45 @@ Docker::imageRootDir = (image) ->
 					else
 						throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 	)
+
+EEXIST = code: 'EEXIST'
+ignore = ->
+MIN_PAGE_SIZE = 4096
+
+aufsMountWithDisposer = (target, layerDiffPaths) ->
+	# We try to create the target directory.
+	# If it exists, it's *probably* from a previous run of this same function,
+	# and the mount will fail if the directory is not empty or something's already mounted there.
+	fs.mkdirAsync(target)
+	.catch code: EEXIST, ignore
+	.then ->
+		appendFromIndex = null
+		options = 'noxino,ro,br='
+		remainingBytes = MIN_PAGE_SIZE - options.length
+		for path, i in layerDiffPaths
```

Comment:
A pure cs version of the previous lodash:
```coffee
layerDiffPaths = layerDiffPaths.map (path) ->
	return "#{path}=ro+wh"
appendFromIndex = layerDiffPaths.findIndex (path) ->
	remainingBytes -= path.length + 1
	# >= -1 because if this is the last entry we won't actually add the comma
	return remainingBytes >= -1
initialLayerPaths = layerDiffPaths[...appendFromIndex]
appendLayerPaths = layerDiffPaths[appendFromIndex...]
options += initialLayerPaths.join(':')
...
Promise.mapSeries [appendLayerPaths], (path) ->
	execAsync("mount -t aufs -o 'remount,append:#{path}' none #{target}")
```